### PR TITLE
More bottom sheet improvements

### PR DESCRIPTION
**Pull request by ivaniskandar**

### CODE REVIEW

**Comment by ivaniskandar** [comment]
[Background dim anim](https://user-images.githubusercontent.com/12537387/119501997-a1f59300-bd93-11eb-911e-766c357719eb.mp4)
[In-out anim](https://user-images.githubusercontent.com/12537387/119502055-b46fcc80-bd93-11eb-94fe-c1345476d9b8.mp4)

File: app/src/main/java/eu/kanade/tachiyomi/widget/sheet/BaseBottomSheetDialog.kt [hunk 1]
```diff
@@ -21,12 +28,34 @@ abstract class BaseBottomSheetDialog(context: Context) : BottomSheetDialog(conte
...
+        sheetBehavior = BottomSheetBehavior.from(bottomSheet)
 
         // Enforce max width for tablets
         val width = context.resources.getDimensionPixelSize(R.dimen.bottom_sheet_width)
         if (width > 0) {
             sheetBehavior.maxWidth = width
         }
+
+        // Set navbar color to transparent for edge-to-edge bottom sheet if we can use light navigation bar
+        // TODO Replace deprecated systemUiVisibility when material-components uses new API to modify status bar icons
```
[the remark this answers is not in the document]
Comment (ivaniskandar) [reply app/src/main/java/eu/kanade/tachiyomi/widget/sheet/BaseBottomSheetDialog.kt:41]: Yeah ideally we should use [WindowsInsetsControllerCompat](https://developer.android.com/reference/androidx/core/view/WindowInsetsControllerCompat?hl=en#setAppearanceLightNavigationBars(boolean)) that uses the new API to set light navigation bar icon, but I guess it doesn't compatible with the deprecated systemuivisibility flags which used by [BottomSheetDialog](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/bottomsheet/BottomSheetDialog.java#L482) to set the light status bar icon.

So at the moment if we use the compat method, the light status bar icons will break on API 30+.

**Comment by arkon** [comment]
I'm actually not sure what the purpose of the animations are. Doesn't it already slide up/down?

**Comment by ivaniskandar** [comment]
Sorry for not doing a proper before-after, that change is mainly to remove the fade animation so that it feels like an actual sheet going in and out. [Current in-out anim](https://user-images.githubusercontent.com/12537387/119595733-33065180-be08-11eb-8aa7-7233a622cbf2.mp4)

We can always set `dismissWithAnimation` to true to use the swipe down animation like when we swipe down the sheet to dismiss but it feels slow. Take a look [here](https://user-images.githubusercontent.com/12537387/119596482-70b7aa00-be09-11eb-81e7-802723cfb8ec.mp4).

**Comment by vengiare** [comment]
> that change is mainly to remove the fade animation so that it feels like an actual sheet going in and out.

I'm pretty sure Google Apps (Google Suite for example) do this fade in/out animation.

**Comment by ivaniskandar** [comment]
I'm sure they do, but I'm not trying to imitate them ¯\_(ツ)_/¯

**Comment by arkon** [comment]
Are you still working on this?

**Comment by ivaniskandar** [comment]
Nope, I guess that's all for now.

**Comment by ivaniskandar** [comment]
image

So there's an issue with the library implementation, thought it only happens on landscape orientation. I have a dirty workaround ready but I think it's better to open an issue on material-components repo and wait for a fix since it's a very minor issue.

**Comment by arkon** [comment]
Related issue: https://github.com/material-components/material-components-android/issues/2221

**Comment by reviewzxy1** [comment]
> Related issue: hey is there any way they'll add comment section in this app, (it's feel lonely)
For Abel to communicate with other readers

**Pull request merged**